akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -369,4 +373,17 @@ object Source {
     new Source(new AcknowledgeSource(bufferSize, overflowStrategy, DefaultAttributes.acknowledgeSource, shape("AcknowledgeSource")))
   }
 
+  /**
+   * Creates a Source from a Files contents.
+   * Emitted elements are `chunkSize` sized [[akka.util.ByteString]] elements,
+   * except the final element, which will be up to `chunkSize` in size.
+   *
+   * You can configure the default dispatcher for this Source by changing the `akka.stream.blocking-io-dispatcher` or
+   * set it for a given Source by using [[ActorAttributes]].
+   *
+   * It materializes a [[Future]] containing the number of bytes read from the source file upon completion.
+   */
+  def file(f: File, chunkSize: Int = 8192): Source[ByteString, Future[Long]] =
```

Comment:
@ktoso Alright, I'll keep the default hardcoded to 8192. Speaking of the demo tho, `Framing` is a terrible name, it's not really about framing, it's about parsing (IMO).

~~~scala
Source.file(thing).via(Parsing.newLines).runForeach(println)
~~~